src/openvic-simulation/economy/trading/GoodMarket.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <memory>
+#include <mutex>
+
+#include "openvic-simulation/economy/trading/BuyUpToOrder.hpp"
+#include "openvic-simulation/economy/trading/MarketSellOrder.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/types/ValueHistory.hpp"
+
+namespace OpenVic {
+	struct GameRulesManager;
+
+	struct GoodMarket {
+	private:
+		static constexpr int32_t exponential_price_change_shift = 7;
+		std::unique_ptr<std::mutex> buy_lock;
+		std::unique_ptr<std::mutex> sell_lock;
+		GameRulesManager const& game_rules_manager;
+		const fixed_point_t base_price;
+		fixed_point_t absolute_maximum_price;
+		fixed_point_t absolute_minimum_price;
+
+		//TODO get from pool instead of storing as field
+		//only used inside execute_orders()
+		std::vector<fixed_point_t> quantity_bought_per_order;
+		std::vector<fixed_point_t> purchasing_power_per_order;
+
+		//only used during day tick (from actors placing order until execute_orders())
+		std::vector<GoodBuyUpToOrder> buy_up_to_orders;
+		std::vector<GoodMarketSellOrder> market_sell_orders;
+
+		protected:
+		bool PROPERTY(is_available);
+
+		private:
```

Comment:
```suggestion
	protected:
		bool PROPERTY_ACCESS(is_available, protected);

	private:
```
- Indentation
- While it technically doesn't matter here as we switch back to `private` straight after, it's best to use `PROPERTY_ACCESS` to ensure the accessibility is still `protected` after declaring `is_available`. `PROPERTY` on its own changes the accessibility to `private`, and so can lead to accidental incorrect accesibility settings

Reply:
Not a fan of PROPERTY_ACCESS since it does not alter the access for the property but only for what comes after it.
I couldn't find a better (and neither could gemini) so we're stuck with this it.